tools/run_tests/python_utils/port_server.py

```diff
@@ -182,4 +186,16 @@ def do_GET(self):
 class ThreadedHTTPServer(ThreadingMixIn, HTTPServer):
   """Handle requests in a separate thread"""
 
+def _flush_output_forever():
+  while True:
+    time.sleep(1)
+    sys.stderr.flush()
+    sys.stdout.flush()
+
+output_flushing_thread = threading.Thread(target=_flush_output_forever)
+output_flushing_thread.daemon = True
+output_flushing_thread.start()
+
+dns_server.start_local_dns_server_in_background(_DNS_SERVER_PORT)
```

Comment:
I think port_server.py  also starting a DNS server as a side effect is a pretty dirty solution (no one would expect such behavior from something called port_server). Also, port server is started in other circumstances - like in performance benchmarks, where you don't need the DNS server at all. I think overall this solution will lead to problems in the future.